bin/05_count_to_api.py

```diff
@@ -1,20 +1,52 @@
 """ Count persons on Prinzipalmarkt and publish to API. """
 
 import requests
+from traffic_cam import classifier, io, predictor, paths
+from time import sleep
 
 
-# download image
-# TODO
+def loop(classifier_model, predictor_model):
+    # download image
+    path = io.download_image()
 
-# classify image location
-# TODO
-# if image location not useful, wait and download new image
-# TODO
+    # classify image location
+    location = classifier.move_image_by_class(source_path=path, model=classifier_model)
 
-# count persons on image
-# TODO
+    # if image location not useful, wait and download new image
+    if "street" not in location:
+        print("Not a street, skipping.")
+        return False
 
-# collect addiational information for API
-# TODO
-# send count to API
-# TODO
+    # count persons on image
+    persons = predictor_model.predict_image(image_path=paths.TRAIN_DIR / location / path.name, plot=True)
+
+    # collect additional information for API
+    rain = True  # 90 % accuracy for Münster, good enough
+
+    # build response
+    response = {
+        "count": persons,
+        "timestamp": io.get_timestamp_isoformat(),
+        "device_id": location,
+        "data": {
+            "rain": "raining" if rain else "not raining",
+        }
+    }
+    print(response)
+
+    # send count to API
+    response = requests.post(
+        "https://counting-backend.codeformuenster.org/counts/",
+        json=response,
+    )
+
+
+if __name__ == "__main__":
```

Comment:
I would suggest to save us the indentation here, and integrate the sleep within the loop.

Reply:
Maybe add the location to the data? The other teams, also do this. Just to be consistent.
`            "longitude": "51.962077",
            "latitude": "7.628095"`

Reply:
Still undecided if that makes sense for a static location, as that info is available for the device already https://counting-backend.codeformuenster.org/devices/south_street_cityhall

Reply:
I suppose oskar knows what frontend needs

Reply:
I would keep sleep outside because it feels very much like a side effect to me ;)